assets/src/domain/eventEditor/ui/datetimes/datesList/actionsMenu/dropdown/DateMainMenu.tsx

```diff
@@ -1,29 +1,39 @@
 import React from 'react';
 import { __ } from '@wordpress/i18n';
 
+import { Copy } from '@application/ui/layout/entityActionsMenu/entityMenuItems';
 import { DropdownMenu, DropdownToggleProps } from '@application/ui/layout';
-
-import CopyDate from './CopyDate';
-import EditDate from './EditDate';
-import TrashDate from './TrashDate';
-import { useDatetimeItem } from '@edtrServices/apollo';
+import { Edit } from '@application/ui/layout/entityActionsMenu/entityMenuItems';
+import { Trash } from '@application/ui/layout/entityActionsMenu/entityMenuItems';
+import { useConfirmationDialog } from '@application/ui/display/confirm';
 
 import { DateMainMenuProps } from './types';
+import useActions from './useActions';
+
+const DateMainMenu: React.FC<DateMainMenuProps> = ({ datetime }) => {
+	const { copyDate, editDate, trashDate, trashed } = useActions({ datetimeId: datetime.id });
+
+	const confirmText = __('Are you sure you want to delete this?');
+	const { confirmationDialog, onOpen } = useConfirmationDialog({
+		confirmText,
+		onConfirm: trashDate,
+	});
 
-const DateMainMenu: React.FC<DateMainMenuProps> = ({ datetime: entity }) => {
-	// Make sure to subscribe to Apollo cache
-	// to avoid stale data
-	const datetime = useDatetimeItem({ id: entity.id });
```

Comment:
There was a reason for this code to be there and it was written in the comment.

Reply:
that was move here, it wasn't deleted:

![image](https://user-images.githubusercontent.com/1477453/82038777-b7c67780-96ac-11ea-8f57-88c909354913.png)

Reply:
Oh, Sorry I didn't see that, thanks :)